### DISCUSSION TITLE
Microsoft outlook incompatibility?

### DISCUSSION THREAD

**Discussion by pitchedwhite** [dngray/ghacks-thunderbird-user.js#40] [General]
I downloaded the user.js version v115.0 and after i downloaded it have been having issues, specifically that i need to enable java script before i can look into office/outlook365. Can anyone help me?

**Comment by pitchedwhite** [comment]
from what i checked, the user.js disabled java script, i still want the user.js installed for privacy reasons, do i either just remove the entire custom user.js or do i change some settings in there? I need help.

**Comment by pitchedwhite** [comment]
theres atleast a hundred java commands in the advanced preferences system config, is there a minimum number that i need to enable to get it working? Or do i need to enable the main file that simply says javascript.enabled true or false. Or am i looking at the wrong place entirely?

**Comment by pitchedwhite** [comment]
found the solution on wiki, should've read that first before opening a thread but i was rushing. The solution was to just change the value of network.cookie.cookieBehavior to 1. Default value was 2. Also it gives the option of changing javascript.enabled value to TRUE instead of FALSE. The thing is though, changing the javascript value did not change whether outlook worked and synced emails or not, it was just the network.cookie.cookieBehavior value being 1 that impacted anything, also i need to keep it turned on indefinitely and can't revert the value back to 2 or else it breaks outlook again. Java for me though seems fine to ignore. Would love to know whether changing the network.cookie.cookieBehavior to 1 and not reverting it back to the value 2 compromises the privacy of my thunderbird or not or has minimal impact on privacy.

**Comment by HorlogeSkynet** [comment]
Hello @pitchedwhite, thanks for your last message and the write-up.
I guess `javascript.enabled` had to be enabled in the past as Microsoft login page required JavaScript. Maybe now it also supports "simple" POST requests without captcha or more mandatory client-side complexity.
I don't know whether we should remove it from the Wiki page (maybe your case was particular ?).

About `network.cookie.cookieBehavior`, you cannot revert it as it controls access to cookies, and they are continuously used against mail servers for authentication.

Bye :wave:
